maro/simulator/scenarios/supply_chain/business_engine.py

```diff
@@ -129,13 +209,104 @@ def _build_world(self) -> None:
 
         self.world.build(conf, self.calc_max_snapshots(), self._max_tick)
 
+    def _build_dag(self) -> None:
+        """Build the task DAG that will be used in `_on_action_received`
+        """
+
+        # Get all consumer units & manufacture units of the world.
+        # They will be reused in `_on_action_received`.
+        self._consumer_units: List[ConsumerUnit] = []  # For reuse
+        self._manufacture_units: List[ManufactureUnit] = []  # For reuse
+        for unit in self.world.get_units_by_root_type(ConsumerUnit):
+            assert isinstance(unit, ConsumerUnit)
+            self._consumer_units.append(unit)
+        for unit in self.world.get_units_by_root_type(ManufactureUnit):
+            assert isinstance(unit, ManufactureUnit)
+            self._manufacture_units.append(unit)
+
+        # Dummy task. Used to complete the DAG's structure
+        # Executing ACTIONS_PROCESS_DONE means all consumer actions are processed.
+        self._dag_task_scheduler.add_task(ACTIONS_PROCESS_DONE)
+
+        # Process consumer actions
+        for unit in self._consumer_units:
+            task_name = TASK_CONSUMER_ACTION_TEMPLATE.format(unit.id)
+            self._dag_task_scheduler.add_task(task_name, unit.process_actions)
```

Comment:
Here the `process_actions` and `process_action` is similar to `on_action_received` before?

I'm wondering whether to change as the implementation now or keep the previous name and let it be as an interface of `UnitBase`